{{cookiecutter.project_slug}}/fabfile.py

```diff
@@ -1,144 +1,233 @@
 
+    def run_in_project_root(self, cmd, **kwargs):
+        """
+        Run command after a cd to the project_root
+        """
+        with self.cd(self.project_root):
+            return self.run(cmd, **kwargs)
 
-def run_pip(args):
-    """
-    Run the pip command in the remote virtualenv.
-    """
-    return run_in_virtualenv("pip", args)
-
-
-def run_python(args):
-    """
-    Run the python command in the remote virtualenv.
-    """
-    return run_in_virtualenv("python", args)
-
-
-def install_requirements():
-    """
-    Install the requirements from the base.txt file to the remote virtualenv.
-    """
-    with cd(get_project_root()):
-        run_pip("install -r requirements/base.txt")
+    def git(self, gitcmd, **kwargs):
+        """
+        git from the project_root
+        """
+        return self.run_in_project_root("git {}".format(gitcmd), **kwargs)
 
+    def run_in_venv(self, cmd, args, **run_kwargs):
+        """
+        Binaries from the venv
+        """
+        return self.run_in_project_root(
+            "{} {}".format(os.path.join(self.venv_path, "bin", cmd), args), **run_kwargs
+        )
 
-def migrate_database():
-    with cd(get_project_root()):
-        run_python("manage.py migrate")
-
+    def mk_venv(self, **run_kwargs):
+        """
+        Create the venv
+        """
+
+        with self.cd(self.site_root):
+            self.run("python3 -m venv venv", **run_kwargs)
+
+    def pip(self, args, **run_kwargs):
+        """
+        pip from the venv, in the project_root
+        """
+        return self.run_in_venv("pip", args, **run_kwargs)
+
+    def python(self, args, **run_kwargs):
+        """
+        python from the venv, in the project_root
+        """
+        return self.run_in_venv("python", args, **run_kwargs)
+
+    def manage_py(self, args, **run_kwargs):
+        """
+        manage.py with the python from the venv, in the project_root
+        """
+        try:
+            env = {
+                "DJANGO_SETTINGS_MODULE": self.config.settings["DJANGO_SETTINGS_MODULE"]
+            }
+        except KeyError:
+            env = {}
+        return self.python("./manage.py {}".format(args), env=env, **run_kwargs)
+
+    def set_setting(self, name, value=None, force: bool = True):
+        """
+        Set a setting in the environment directory, for use by Django
+        """
+        envfile_path = os.path.join(self.envdir_path, name)
+
+        will_write = force
+        if not force:
+            try:
+                # Test that it does exist
+                self.run_in_project_root("test -r {}".format(envfile_path), hide=True)
+            except UnexpectedExit:
+                will_write = True
+
+        if will_write:
+            if value is None:
+                value = input("Value for {}: ".format(name))
+
+            # Convert booleans into ints, so that Django takes them back easily
+            if isinstance(value, bool):
+                value = int(value)
```

Comment:
Unfortunately this won't be enough because of the way we parse boolean environment variables in settings. We only consider empty or non-existent variables as `False`.
E.g. : https://github.com/liip/django-template/blob/986f728aeb6f55e4aa39fa2758fc2fffe1e5b622/%7B%7Bcookiecutter.project_slug%7D%7D/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/dev.py#L4
`bool("0") == True`

But maybe we should change the way we parse these variables ?